### ISSUE TITLE
PHPMailer support in PHP 7.4.x

### ISSUE THREAD

**Issue by artur-stepien** [joomla/joomla-cms#28395]
### Steps to reproduce the issue
Use any extension that sends attachments using Joomla! mailer. Sending any attachment via Joomla mailer (that uses phpmailer) in PHP 7.4+ will cause a deprecated warning as follows:
```
Deprecated: Function get_magic_quotes_runtime() is deprecated in /var/www/libraries/vendor/phpmailer/phpmailer/class.phpmailer.php on line 2714
```
Or just look into `/libraries/vendor/phpmailer/phpmailer/class.phpmailer.php` file on line `2714` and check `get_magic_quotes_runtime()` documentation in [https://www.php.net/manual/en/function.get-magic-quotes-runtime.php](https://www.php.net/manual/en/function.get-magic-quotes-runtime.php "https://www.php.net/manual/en/function.get-magic-quotes-runtime.php"). This function is deprecated starting from 7.4.0

### Expected result
No warning

### Actual result
```
Deprecated: Function get_magic_quotes_runtime() is deprecated in /var/www/libraries/vendor/phpmailer/phpmailer/class.phpmailer.php on line 2714
```

### System information (as much as possible)
PHP 7.4.3
Joomla 3.9.16

**Comment by brianteeman** [comment]
Thanks for reporting this but this is a vendor supplied library and you should report it directly to them

https://github.com/PHPMailer/PHPMailer/

**Comment by artur-stepien** [comment]
Its not them who put this version in Joomla 3.9. That's first thing. Second, they already have a 6.x that removed the deprecated code. Joomla! 4.x also uses PHPMailer 6.x and until PHPMailer is updated in Joomla 3.9.x it will cause deprecated warnings in some of extensions when running on PHP 7.4.

**Comment by mbabker** [comment]
It is not Joomla’s responsibility to fork and patch third party libraries that it uses, therefore the instruction to report the issue upstream is valid.

The library cannot be updated because of the raised PHP minimum not matching Joomla’s.

A deprecation notice will not break your site until the deprecated functionality is removed.  It can be safely ignored.

**Comment by artur-stepien** [comment]
@mbabker Where did I wrote anything about patching external libraries? Point me to that sentence. And please, read the comment above, they did patch it in next release, there is nothing to report. The issue exists and ignoring the deprecated warning is not the solution is exactly the lack of one. If I make a PR to PHPMailer and they will release another version of 5.x will you update the library in Joomla?

**Comment by gerryfrancis** [comment]
Technical requirements for Joomla! are PHP 7.3+, that would include PHP 7.4. Any components included in current versions of Joomla! should support that, otherwise the specs must be defined more precisely.

**Comment by artur-stepien** [comment]
PHPMailer just released 5.2.28 that should fix the issue. So could you kindly update the version Joomla!  uses? @mbabker @brianteeman

**Comment by brianteeman** [comment]
@gerryfrancis that is the requirements for joomla 4 not 3

@artur-stepien feel free to submit a pull request

**Comment by infograf768** [comment]
Is 3.9.x not supposed to work with php 7.4?

**Comment by brianteeman** [comment]
yes and it does

**Comment by infograf768** [comment]
Therefore it makes sense to update phpmailer.
Reopening Issue.

**Comment by infograf768** [comment]
Except evidently if it would break for lower php versions...

**Comment by artur-stepien** [comment]
@infograf768 No it would not, there is a new version of PHPMailer with just fixing the deprecated issue reported above. Its still from 5.x

**Comment by wilsonge** [comment]
> PHPMailer just released 5.2.28 that should fix the issue. So could you kindly update the version Joomla! uses?

Yes we can - here's a pull request https://github.com/joomla/joomla-cms/pull/28398

Thankyou for reporting it :)
